cli-plugins/manager/manager.go

```diff
@@ -0,0 +1,190 @@
+package manager
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"sync"
+	"syscall"
+
+	"github.com/docker/cli/cli/command"
+	"github.com/spf13/cobra"
+)
+
+// ErrPluginNotFound is the error returned when a plugin could not be found.
+type ErrPluginNotFound string
+
+func (e ErrPluginNotFound) Error() string {
+	return "Error: No such CLI plugin: " + string(e)
+}
+
+var (
+	pluginDirs     []string
+	pluginDirsOnce sync.Once
+)
+
+func getPluginDirs(dockerCli command.Cli) []string {
```

Comment:
Just wondering, what about running this piece of code in an `init()` function instead of using this `sync.Once` ?

Reply:
Looking at this I'm not sure why this needs to be done exactly once, I don't think it's an especially hot path, it's not terribly expensive and I think the answer is idempotent in a given process (unless someone was futzing with its environment). Not sure what I was thinking. I think I should just nuke the sync stuff. WDYT?

Reply:
I'm fine with the nuke 💥and compute it everytime.